src/rard/research/views/mention.py

```diff
@@ -26,102 +27,145 @@ class MentionSearchView(LoginRequiredMixin, View):
+            "af": self.anonymous_fragment_search,
+            "fr": self.fragment_search,
+            "bi": self.bibliography_search,
+            "uf": self.unlinked_fragment_search,
+            "wk": self.work_search,
+        }
+
+    def order_number_search(keywords):
+        order_number = None
+        for kw in keywords:
+            if kw.isnumeric():
+                order_number = int(kw)
+                keywords.remove(kw)
+        return order_number, keywords
 
     @classmethod
-    def topic_search(cls, keywords):
-        qs = Topic.objects.all()
-        results = qs.filter(name__icontains=keywords)
-        return results.distinct()
+    def basic_search(cls, self, method, keywords):
+        target_model = self.BASIC_SEARCH_TYPES[method][0]
+        filter_slug = self.BASIC_SEARCH_TYPES[method][1]
+        ordering = self.BASIC_SEARCH_TYPES[method][2]
+
+        model_query = Q()
+        for kw in keywords:
+            model_query = model_query & Q(**{filter_slug: kw})
+
+        qs = target_model.objects.all()
+        results = qs.filter(model_query)
+
+        return results.distinct().order_by(ordering)
 
     @classmethod
-    def work_search(cls, keywords):
-        qs = Work.objects.all()
-        results = qs.filter(name__icontains=keywords)
+    def anonymous_fragment_search(cls, keywords):
+        # do we want to make it so the search is inclusive, not exact?
+        # i.e @af:1 returns 1,10,11,etc
+        order_number = cls.order_number_search(keywords)[0]
+        qs = AnonymousFragment.objects.all()
+        if order_number:
+            order_number = order_number - 1
+            order_query = Q(id=order_number) | Q(order=order_number)
+        else:
+            order_query = Q()
+
+        results = qs.filter(order_query)
         return results.distinct()
 
     @classmethod
-    def fragment_search(cls, keywords):
-        qs = Fragment.objects.all()
-        results = qs.filter(
-            antiquarian_fragmentlinks__antiquarian__name__icontains=keywords  # noqa
+    def bibliography_search(cls, keywords):
+        qs = BibliographyItem.objects.annotate(
+            author_title=Concat(
+                F("authors"), Value(" "), F("title"), output_field=CharField()
+            )
         )
+        bib_query = Q()
+        for kw in keywords:
+            bib_query = bib_query & Q(author_title__icontains=kw)
+
+        results = qs.filter(bib_query)
         return results.distinct()
 
     @classmethod
-    def remove_keyword(cls, kw, keywords):
-        if not keywords or not keywords[0].lower().startswith(kw):
-            return None
-        k = keywords[0][len(kw) :]
-        if len(k) == 0:
-            return keywords[1:]
-        keywords[0] = k
-        return keywords
+    def work_search(cls, keywords):
+        qs = Work.objects.annotate(
+            author_title=Concat(
+                F("authors"), Value(" "), F("name"), output_field=CharField()
+            )
+        )
```

Comment:
I noticed authors just uses the antiquarian's order name (e.g. Accius for Lucius Accius). That means if you do a work search for `@wk:lucius` it won't show any of Accius's works. We can get the whole name by copying the pattern used by the WorkManager which uses StringAgg to add the authors annotation, but use antiquarian__name instead.

```suggestion
        qs = Work.objects.annotate(
            author_title=Concat(
                StringAgg("antiquarian__name", delimiter=","),
                Value(" "),
                F("name"),
                output_field=CharField()
            )
        )
```

Reply:
oh nice, thanks!